.github/workflows/master.yml

```diff
@@ -0,0 +1,56 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2024 Intel Corporation
+
+name: Test integration
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build
+        run: |
+          go build -o build/_output/onos ./cmd/onos
+          go build -o build/_output/onos-cli-docs-gen ./cmd/onos-cli-docs-gen
+          go build -o build/_output/gnmi_cli ./cmd/gnmi_cli
+
+  mod-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: mod lint
+        run: |
+          go mod tidy
+          go mod vendor
+          bash -c "diff -u <(echo -n) <(git diff go.sum)"
```

Comment:
This whole action is not quite clear to me. I do not understand what is the benefit of the comparison. Do you know why this is done or has to be done?

Reply:
This the check that all the dependencies are in place

Reply:
I do not see this as useful because that is something that the developer should do